sdlib/d/gc/extent.d

```diff
@@ -266,7 +293,59 @@ public:
 		// FIXME: in contract.
 		assert(isSlab(), "slabData accessed on non slab!");
 
-		return _metadata.slabData;
+		return _metadata.slabData.slabData;
+	}
+
+	/**
+	 * Freespace Flag features for slabs.
+	 */
+
+	@property
+	bool supportsFreeSpace() const {
+		return isAppendableSizeClass(sizeClass);
+	}
+
+	@property
+	ref shared(Bitmap!256) freeSpaceFlags() {
+		assert(isSlab(), "freeSpaceFlags accessed on non slab!");
+		assert(supportsFreeSpace, "freeSpaceFlags not supported!");
+
+		return _metadata.slabData.freeSpaceData.freeSpaceFlags;
+	}
+
+	bool setFreeSpace(uint index, size_t freeSpace) {
+		assert(isSlab(), "setFreeSpace accessed on non slab!");
+		assert(freeSpace <= slotSize, "freeSpace exceeds alloc size!");
+		assert(index < slotCount, "index is out of range!");
+
+		if (!supportsFreeSpace) {
```

Comment:
In what situation do we expect the caller to call on an unsupported size class, and that wouldn't be a bug?

We ought to select a suitable size class when allocating appendable, and when extending, we'll always do a new allocation when we have an unsuitable size class.

Reply:
Do we need to set the free space in realloc?